src/ghga_connector/core/uploading/uploader.py

```diff
@@ -12,362 +12,144 @@
+                match the expected value.
+            CompleteFileUploadError: If there's an error completing the file upload.
         """
-        if not self._upload_id:
-            raise exceptions.UploadIdUnsetError()
-
-        # build url and headers
-        url = f"{self._api_url}/uploads/{self._upload_id}"
-        headers = {"Accept": "*/*", "Content-Type": "application/json"}
-        post_data = {"status": upload_status}
-        serialized_data = json.dumps(post_data)
-
-        try:
-            response = await self._client.patch(
-                url=url, headers=headers, content=serialized_data
-            )
-        except httpx.RequestError as request_error:
-            exceptions.raise_if_connection_failed(request_error=request_error, url=url)
-            raise exceptions.RequestFailedError(url=url) from request_error
+        num_parts = calc_number_of_parts(
+            self._encryptor.expected_encrypted_size, self._part_size
+        )
+        self._in_sequence_part_number = 1
+
+        # Encrypt and upload file parts in parallel
+        self._progress_bar = self.new_progress_bar()
+        with self._file_path.open("rb") as file, self._progress_bar:
+            file_processor = self._encryptor.process_file(file=file)
+            task_handler = TaskHandler()
+            for _ in range(num_parts):
+                task_handler.schedule(
+                    self._upload_file_part(file_processor=file_processor)
+                )
+            # Wait for all upload tasks to finish
+            await task_handler.gather()
 
-        status_code = response.status_code
-        if status_code != 204:
-            spec = {
-                400: {
-                    "uploadNotPending": lambda: exceptions.CantChangeUploadStatusError(
-                        upload_id=self._upload_id, upload_status=upload_status
-                    ),
-                    "uploadStatusChange": lambda: exceptions.CantChangeUploadStatusError(
-                        upload_id=self._upload_id, upload_status=upload_status
-                    ),
-                },
-                403: {
-                    "noFileAccess": lambda: exceptions.NoUploadAccessError(
-                        upload_id=self._upload_id
-                    )
-                },
-                404: {
-                    "noSuchUpload": lambda: exceptions.UploadNotRegisteredError(
-                        upload_id=self._upload_id
-                    )
-                },
-            }
-            ResponseExceptionTranslator(spec=spec).handle(response=response)
-            raise exceptions.BadResponseCodeError(url=url, response_code=status_code)
+        # Get the unencrypted checksum and tell the Upload API to conclude the S3 upload
+        unencrypted_checksum = self._encryptor.checksums.get()[0]
+        encrypted_checksum = self._encryptor.checksums.encrypted_checksum_for_s3()
```

Comment:
Any other place where the `encrypted_checksum` is actually retrieved from `checksums.get()`?
If not, the access could we slightly simplified and the index removed here